src/Plugin.cpp

```diff
@@ -36,40 +36,64 @@ namespace csp::atmospheres {
+  cs::core::Settings::deserialize(j, "rayleighAnisotropy", o.mRayleighAnisotropy);
+  cs::core::Settings::deserialize(j, "cloudTexture", o.mCloudTexture);
+  cs::core::Settings::deserialize(j, "cloudHeight", o.mCloudHeight);
+}
+
+void to_json(nlohmann::json& j, Plugin::Settings::Atmosphere const& o) {
+  cs::core::Settings::serialize(j, "atmosphereHeight", o.mAtmosphereHeight);
+  cs::core::Settings::serialize(j, "mieHeight", o.mMieHeight);
+  cs::core::Settings::serialize(j, "mieScatteringR", o.mMieScatteringR);
+  cs::core::Settings::serialize(j, "mieScatteringG", o.mMieScatteringG);
+  cs::core::Settings::serialize(j, "mieScatteringB", o.mMieScatteringB);
+  cs::core::Settings::serialize(j, "mieAnisotropy", o.mMieAnisotropy);
+  cs::core::Settings::serialize(j, "rayleighHeight", o.mRayleighHeight);
+  cs::core::Settings::serialize(j, "rayleighScatteringR", o.mRayleighScatteringR);
+  cs::core::Settings::serialize(j, "rayleighScatteringG", o.mRayleighScatteringG);
+  cs::core::Settings::serialize(j, "rayleighScatteringB", o.mRayleighScatteringB);
+  cs::core::Settings::serialize(j, "rayleighAnisotropy", o.mRayleighAnisotropy);
+  cs::core::Settings::serialize(j, "cloudTexture", o.mCloudTexture);
+  cs::core::Settings::serialize(j, "cloudHeight", o.mCloudHeight);
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void from_json(const nlohmann::json& j, Plugin::Settings& o) {
-  cs::core::parseSection("csp-atmospheres", [&] {
-    o.mAtmospheres =
-        cs::core::parseMap<std::string, Plugin::Settings::Atmosphere>("atmospheres", j);
-  });
+void from_json(nlohmann::json const& j, Plugin::Settings& o) {
+  cs::core::Settings::deserialize(j, "atmospheres", o.mAtmospheres);
+  cs::core::Settings::deserialize(j, "enabled", o.mEnabled);
+  cs::core::Settings::deserialize(j, "quality", o.mQuality);
+  cs::core::Settings::deserialize(j, "waterLevel", o.mWaterLevel);
+  cs::core::Settings::deserialize(j, "enableClouds", o.mEnableClouds);
+  cs::core::Settings::deserialize(j, "enableLightShafts", o.mEnableLightShafts);
+  cs::core::Settings::deserialize(j, "enableWater", o.mEnableWater);
+}
+
+void to_json(nlohmann::json& j, Plugin::Settings const& o) {
+  cs::core::Settings::serialize(j, "atmospheres", o.mAtmospheres);
+  cs::core::Settings::serialize(j, "enabled", o.mEnabled);
+  cs::core::Settings::serialize(j, "quality", o.mQuality);
+  cs::core::Settings::serialize(j, "waterLevel", o.mWaterLevel);
+  cs::core::Settings::serialize(j, "enableClouds", o.mEnableClouds);
+  cs::core::Settings::serialize(j, "enableLightShafts", o.mEnableLightShafts);
+  cs::core::Settings::serialize(j, "enableWater", o.mEnableWater);
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 Plugin::Plugin()
-    : mProperties(std::make_shared<Properties>()) {
+    : mPluginSettings(std::make_shared<Settings>()) {
```

Comment:
Can this be a `std::unique_ptr`? It will never leave this plugin if I'm correct.

Reply:
In fact, it doesn't have to be pointer at all. Good catch!